nav-app/src/app/techniques-search/techniques-search.component.ts

```diff
@@ -37,71 +39,140 @@ export class TechniquesSearchComponent implements OnInit {
-                let c2 = tB.isSubtechnique ? tB.parent.name : tB.name;
-                return c1.localeCompare(c2)
-            });
-            // console.log(seenIDs, results)
-            this.results = results;
-        } else {
-            this.results = [];
+    getTechniques() {
+        //get master list of techniques and sub-techniques
+        let allTechniques = this.dataService.getDomain(this.viewModel.domainID).techniques;
+        for (let technique of allTechniques) {
+            allTechniques = allTechniques.concat(technique.subtechniques);
         }
+        this.techniqueResults = this.filterAndSort(allTechniques, "", true);
     }
-    public get queryLength(): number {
-        return this._query.length;
+
+    getTechniqueResults(query = "") {
+        if (query.trim() != "" && query.includes(this.previousQuery)) {
+            this.techniqueResults = this.filterAndSort(this.techniqueResults, query, true);
+        } else {
+            this.getTechniques();
+        }
     }
-    public results: Technique[] = [];
 
-    constructor(private dataService: DataService) {
+    getStixData() {
+        let domain = this.dataService.getDomain(this.viewModel.domainID);
 
+        this.stixTypes = [{
+            "label": "threat groups",
+            "objects": this.filterAndSort(domain.groups)
+        }, {
+            "label": "software",
+            "objects": this.filterAndSort(domain.software)
+        }, {
+            "label": "mitigations",
+            "objects": this.filterAndSort(domain.mitigations)
+        }]
     }
 
-    ngOnInit() {
+    // getStixResults() checks if query is:
+    // 1) valid, and
+    // 2) part of last query, otherwise call getStixData() to search all objects again
+    getStixResults(query = "") {
+        if (query.trim() != "" && query.includes(this.previousQuery)) {
+            this.stixTypes.forEach(item => item['objects'] = this.filterAndSort(item['objects'], query));
+        } else {
+            this.getStixData();
+        }
     }
 
     public toggleFieldEnabled(field: string) {
+        let temp = this._query;
         for (let thefield of this.fields) {
             if (thefield.field == field) {
                 thefield.enabled = !thefield.enabled;
+                // this is to trigger getTechniques() and getStixData() in the case that:
+                // a field was toggled, and
+                // the query did not change
+                this.query = "";
                 break;
             }
         }
+        this._query = temp;
         this.query = this._query;
     }
 
+    public mouseEnter(technique: Technique, isStix?: boolean): void {
+        if (isStix) {
+            for (let t of this.getRelated(technique)) {
+                this.viewModel.selectTechniqueAcrossTactics(t, true, true);
+            }
+        } else {
+            this.viewModel.highlightTechnique(technique);
+        }
+    }
```

Comment:
For the purposes of correctness `isStix` should be changed. Techniques are themselves STIX like all other objects. `isNonTechnique` (or `isTechnique` while inverting the if statement) would be better.